### DISCUSSION TITLE
Separate nodeADC.login and NodeADC.getCurrentState possible?

### DISCUSSION THREAD

**Discussion by shicks3** [node-alarm-dot-com/node-alarm-dot-com#51] [General]
This is more a question, not an issue.
I've integrated nodeADC into my MagicMirror install.  It is working well but I have a couple of questions:
1. Can the login() and and getCurrentState() methods be separated?  I've been trying different tests including capturing the authOpts from login() and keeping it to pass to getCurrentState().  I only receive login failed messages.   Might be something simple or my lack of understanding of the promise structure?

The idea is to call login once (and when needed) and then to call getCurrentState as often as I want for updating the display.  Another google project does this separately to reduce login calls.   It keeps the login information until the data calls return and error.   Then it logs in again. 

This is the original (from your readme)
	nodeADC
		.login(config.ADCusername, config.ADCpassword, config.existingMfaToken)
		.then((authOpts) => nodeADC.getCurrentState(authOpts.systems[0], authOpts))
		.then((res) => {
			console.log('Res:', JSON.stringify(res));
               });

2. is this idea even necessary?  I'm not finding anywhere that shows how often permissible to login so maybe this isn't necessary. 
Thanks for any ideas you might have.

**Comment by chase9** [comment]
I'm sure you've moved on by now, but for anyone needing this in the future, login() and getCurrentState() are different methods and are not needed to be linked. However, getCurrentState() does take authOpts, which is returned by login.

We do something similar to what you are trying to do in homebridge-node-alarm-dot-com and usually refresh every 30 minutes to 1 hour. If you're using websockets, they appear to invalidate the auth token every 5 minutes.
